packages/material-ui-lab/src/Masonry/Masonry.js

```diff
@@ -81,48 +120,79 @@ const Masonry = React.forwardRef(function Masonry(inProps, ref) {
   });
 
   const masonryRef = React.useRef();
+  const [maxColumnHeight, setMaxColumnHeight] = React.useState();
+  const [maxNumOfRows, setMaxNumOfRows] = React.useState();
   const { children, className, component = 'div', columns = 4, spacing = 1, ...other } = props;
-  const ownerState = { ...props, spacing, columns };
+  const ownerState = { ...props, spacing, columns, maxColumnHeight, maxNumOfRows };
   const classes = useUtilityClasses(ownerState);
+  const lineBreakStyle = {
+    flexBasis: '100%',
+    width: 0,
+    margin: 0,
+    padding: 0,
+    content: '',
+  };
 
-  const contextValue = React.useMemo(() => ({ spacing }), [spacing]);
-  let didWarn = false;
   React.useEffect(() => {
-    // scroller always appears when masonry's height goes beyond 2,000px on Chrome
-    const handleScroll = () => {
-      if (masonryRef.current.clientHeight === 1998 && !didWarn) {
-        console.warn(
-          [
-            'Material-UI: The Masonry can have the maximum height of 2,000px on Chrome browser.',
-            'Items that go beyond this height fail to be rendered on Chrome browser.',
-            'You can find more in this open issue: https://github.com/mui-org/material-ui/issues/27934',
-          ].join('\n'),
-        );
-
-        // eslint-disable-next-line react-hooks/exhaustive-deps
-        didWarn = true;
+    const handleResize = () => {
+      const columnHeights = [];
+      const numOfRows = [];
+      let skip = false;
+      masonryRef.current.childNodes.forEach((child) => {
```

Comment:
You want to guard against any `Node` other than `Element` by checking `child.nodeType !== Node.ELEMENT_NODE`. If the masonry contains text nodes or comment nodes then the logic in the loop breaks on e.g. `child.dataset.class` with `cannot read property class of undefined`.

Right now it might seem unnecessary but React might choose to insert other nodes between the children. May already be problematic in server rendered React 18 since they use comment nodes for meta data I believe.

Reply:
Sure. Thanks for your detailed feedback!! :)